src/edu/stanford/dlss/wowza/SulWowza.java

```diff
@@ -15,73 +17,203 @@
+        // could validate it's an IP address format, but since we obtain it from http session, we trust it
+        if (userIp != null && userIp.length() >= MIN_IP_LENGTH)
+            return true;
+        else
+        {
+            getLogger().error(this.getClass().getSimpleName() + ": User IP missing or implausibly short" +
+                                (userIp == null ? "" : ": " + userIp));
+            return false;
+        }
+    }
+
+    private static final int DRUID_TREE_LENGTH = "aa/123/bb/4567".length();
+
+    /** require streamName to have format "aa/000/aa/0000/sample.mp4"  (druid tree)/(media_fname) */
+    boolean validateStreamName(String streamName)
     {
-        getLogger().info("Tommy: SulWowza token = " + token);
-        return true;
+        if (streamName == null || streamName.length() <= DRUID_TREE_LENGTH)
+        {
+            getLogger().error(this.getClass().getSimpleName() + ": streamName missing or implausibly short" +
+                                (streamName == null ? "" : ": " + streamName));
+            return false;
+        }
+        String[] pieces = streamName.split("/");
+        if (pieces.length == 5)
+            return true;
+        else
+        {
+            getLogger().error(this.getClass().getSimpleName() + ": unable to parse druid and filename from streamName " + streamName);
+            return false;
+        }
+    }
+
+    static private final Pattern DRUID_REGEX_PATTERN = Pattern.compile("[a-z]{2}\\d{3}[a-z]{2}\\d{4}");
+
+    /** the druid of the media file to be streamed;  needed to verify stacks token
+     * Assumption:  validateStreamName() was already called */
+    String getDruid(String streamName)
+    {
+        String[] pieces = streamName.split("/");
+        String druid = pieces[0] + pieces[1] + pieces[2] + pieces[3];
+        if (DRUID_REGEX_PATTERN.matcher(druid).matches())
+            return druid;
+        else
+            return null;
+    }
+
+    /** the media file to be streamed, as stored under the druid-tree;  needed to verify stacks token
+     * Assumption:  validateStreamName() was already called */
+    String getFilename(String streamName)
+    {
+        String filename = streamName.substring(streamName.lastIndexOf('/') + 1);
+        if (filename.length() > 0)
+            return filename;
+        else
+            return null;
+    }
+
+    /** Assumption: stacksToken, druid, userIp and filename are all reasonable values (non-null, not empty, etc.) */
+    String getVerifyStacksTokenUrl(String stacksToken, String druid, String filename, String userIp)
+    {
+        // TODO:  Url encode anything?   utf-8 charset affect anything? (filename, stacksToken)
+        String queryStr = "stacks_token=" + stacksToken + "&user_ip=" + userIp;
+        String fullUrl = stacksTokenVerificationBaseUrl + "/media/" + druid + "/" + filename + "/verify_token?" + queryStr;
+        try
+        {
+            new URL(fullUrl);
+            return fullUrl;
```

Comment:
How about making this method return a URL instead of a String? Then you wouldn't have to re-instantiate the URL again when you call verifyStacksToken(String) from within verifyStacksToken(String, String, String, String)?